### PR TITLE
Retry lineage query in case of race condition.

### PR DESCRIPTION
**Pull request by CuriousVini**
The reason for this change is there is a race condition in updating the run records and lineage metadata since they are 2 different tables. Now lineage access table creates 2 entries for a dataset: first one with unknown access type and second with read/write access type. This the second entry is not present when integration test queries it, this `PurchaseMetadataTest` can fail. So we should retry for lineage. 

I have tested the fix on the failure cluster.

### CODE REVIEW

File: integration-test-remote/src/test/java/co/cask/cdap/apps/metadata/PurchaseMetadataTest.java [hunk 1]
```diff
@@ -135,6 +135,14 @@ public void testLineage() throws Exception {
...
                        ImmutableSet.of(PURCHASE_FLOW.flowlet("reader")))
         )), ImmutableSet.<CollapseType>of());
+
+    // wait for lineage to be written
+    final LineageRecord lineageRecordExpected = expected;
+    Tasks.waitFor(true, () -> {
+      String actual = String.valueOf(lineageClient.getLineage(PURCHASE_STREAM, startTime, endTime, null));
+      return String.valueOf(lineageRecordExpected).equals(actual);
+    }, 30, TimeUnit.SECONDS, 500, TimeUnit.MILLISECONDS);
+
```
Comment (sagarkapare) [inline integration-test-remote/src/test/java/co/cask/cdap/apps/metadata/PurchaseMetadataTest.java]: I think you can just check on the `expected` value, in Tasks.waitFor.
For example - 
```java
Tasks.waitFor(expected, () -> {
      return lineageClient.getLineage(PURCHASE_STREAM, startTime, endTime, null);
    }, 30, TimeUnit.SECONDS, 500, TimeUnit.MILLISECONDS);
```

File: integration-test-remote/src/test/java/co/cask/cdap/apps/metadata/PurchaseMetadataTest.java [hunk 2]
```diff
@@ -135,6 +135,14 @@ public void testLineage() throws Exception {
...
     Assert.assertEquals(expected, lineageClient.getLineage(PURCHASE_STREAM, startTime, endTime, null));
```
Comment (sagarkapare) [inline integration-test-remote/src/test/java/co/cask/cdap/apps/metadata/PurchaseMetadataTest.java]: This assertion can now be removed.

**Comment by CuriousVini** [comment]
@sagarkapare Addressed comments. Please take another look when you get a chance, thanks!

**Comment by CuriousVini** [comment]
Compiled locally. and ran test on the cluster. Merging it for today's ITN

**Pull request merged**